File: shared/types/gameState.ts
export interface GameState {
  /** Левая строка синей верхней панели (название турнира / лиги) */
  TournamentTitle: string;
  /** Правая строка синей верхней панели (серия, счёт серии и т.п.) */
  SeriesInfo: string;
  /**
   * Имя файла картинки брендинга турнира в `{VITE_BASE_LOGO_URL}/logos/`.
   * Пустая строка — показывается текст «Time of the stars».
   */
  BrandingImage: string;
  TeamA: string;
  TeamAFull: string;
  TeamB: string;
  TeamBFull: string;
  penalty_a: string;
  penalty_b: string;
  ScoreA: number;
  ScoreB: number;
  ShotsA: number;
  ShotsB: number;
  logo_a: string;
  logo_b: string;
  Timer: string;
  /** Таймер большинства (MM:SS); тикает на сервере вместе с Timer, пока Running и PowerPlayActive */
  PowerPlayTimer: string;
  PowerPlayActive: boolean;
  Period: number;
  Running: boolean;
  Visible: boolean;
}

export const defaultGameState: GameState = {
  TournamentTitle: "ТВОЯ ИГРА",
  SeriesInfo: "",
  BrandingImage: "",
  TeamA: "ARC",
  TeamAFull: "ARCTIC",
  TeamB: "ST",
  TeamBFull: "StrongTeam",
  penalty_a: "None",
  penalty_b: "БОЛЬШЕНСТВО 01:36",
  ScoreA: 1,
  ScoreB: 1,
  ShotsA: 14,
  ShotsB: 17,
  logo_a: "team-a.png",
  logo_b: "team-b.png",
  Timer: "11:13",
  PowerPlayTimer: "02:00",
  PowerPlayActive: false,
  Period: 2,
  Running: false,
  Visible: true,
};

File: backend/app/models/game_state.py
from typing import Optional

from pydantic import BaseModel, Field


# class GameState(BaseModel):
#     TournamentTitle: str = "STANLEY CUP PLAYOFFS GEICO."
#     SeriesInfo: str = "SERIES TIED 1-1"
#     BrandingImage: str = ""
#     TeamA: str = "ARC"
#     TeamAFull: str = "ARCTIC"
#     TeamB: str = "ST"
#     TeamBFull: str = "StrongTeam"
#     penalty_a: str = "None"
#     penalty_b: str = "БОЛЬШЕНСТВО 01:36"
#     ScoreA: int = 1
#     ScoreB: int = 1
#     ShotsA: int = 14
#     ShotsB: int = 17
#     logo_a: str = "team-a.png"
#     logo_b: str = "team-b.png"
#     Timer: str = "11:13"
#     PowerPlayTimer: str = "02:00"
#     PowerPlayActive: bool = False
#     Period: int = 2
#     Running: bool = False
#     Visible: bool = True


class GameStatePatch(BaseModel):
    TournamentTitle: Optional[str] = None
    SeriesInfo: Optional[str] = None
    BrandingImage: Optional[str] = None
    TeamA: Optional[str] = None
    TeamAFull: Optional[str] = None
    TeamB: Optional[str] = None
    TeamBFull: Optional[str] = None
    penalty_a: Optional[str] = None
    penalty_b: Optional[str] = None
    ScoreA: Optional[int] = Field(default=None, ge=0)
    ScoreB: Optional[int] = Field(default=None, ge=0)
    ShotsA: Optional[int] = Field(default=None, ge=0)
    ShotsB: Optional[int] = Field(default=None, ge=0)
    logo_a: Optional[str] = None
    logo_b: Optional[str] = None
    Timer: Optional[str] = None
    PowerPlayTimer: Optional[str] = None
    PowerPlayActive: Optional[bool] = None
    Period: Optional[int] = Field(default=None, ge=1)
    Running: Optional[bool] = None
    Visible: Optional[bool] = None
